platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/internal/DefaultJavaFeatureSpec.java

```diff
@@ -82,6 +82,14 @@ public JvmFeatureInternal create() {
             capabilities.add(new ProjectDerivedCapability(project, name));
         }
 
+        if (SourceSet.isMain(sourceSet)) {
+            DeprecationLogger.deprecate("Registering a feature using the main source set")
```

Comment:
🤔 I would add a bit more context like the name and avoid using the word "register".  The API is `usingSourceSet(...)`, so maybe something like:

> Feature '$name' uses the main source set.

Reply:
I used register since the API is called `registerFeature`. I felt for that reason it makes sense to include register in the message